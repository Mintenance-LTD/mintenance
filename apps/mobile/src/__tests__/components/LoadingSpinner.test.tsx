import React from 'react';
import { render } from '@testing-library/react-native';
import {
  LoadingSpinner,
  FullScreenLoading,
} from '../../components/LoadingSpinner';

describe('LoadingSpinner', () => {
  it('renders with default props', () => {
    const { getByTestId } = render(<LoadingSpinner />);

    expect(getByTestId('loading-spinner')).toBeTruthy();
  });

  it('displays custom message when provided', () => {
    const customMessage = 'Loading your jobs...';
    const { getByText } = render(<LoadingSpinner message={customMessage} />);

    expect(getByText(customMessage)).toBeTruthy();
  });

  it('renders without message when message prop is undefined', () => {
    const { queryByText } = render(<LoadingSpinner message={undefined} />);

    expect(queryByText('Loading...')).toBeTruthy(); // default message
  });

  it('applies custom size when provided', () => {
    const { getByTestId } = render(<LoadingSpinner size='large' />);

    const spinner = getByTestId('loading-spinner');
    expect(spinner).toBeTruthy();
  });

  it('applies custom color when provided', () => {
    const { getByTestId } = render(<LoadingSpinner color='#FF0000' />);

    const spinner = getByTestId('loading-spinner');
    expect(spinner).toBeTruthy();
  });

  it('renders FullScreenLoading component', () => {
    const { getByText } = render(<FullScreenLoading />);

    expect(getByText('Loading...')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<LoadingSpinner message='Loading...' />);

    expect(toJSON()).toMatchSnapshot();
  });
});
