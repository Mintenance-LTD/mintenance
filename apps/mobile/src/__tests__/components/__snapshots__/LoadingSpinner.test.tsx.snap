// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoadingSpinner matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 20,
    }
  }
  testID="loading-spinner"
>
  <ActivityIndicator
    color="#007AFF"
    size="large"
  />
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginTop": 15,
          "textAlign": "center",
        },
        {
          "color": "#007AFF",
        },
      ]
    }
    testID="loading-text"
  >
    Loading...
  </Text>
</View>
`;
